pkg/services/secrets/database/database.go

```diff
@@ -16,22 +16,28 @@ import (
 const dataKeysTable = "data_keys"
 
 type SecretsStoreImpl struct {
-	sqlStore db.DB
-	log      log.Logger
+	sb               db.DB
+	log              log.Logger
+	successCallbacks *sync.Map
 }
 
-func ProvideSecretsStore(sqlStore db.DB) *SecretsStoreImpl {
-	return &SecretsStoreImpl{
-		sqlStore: sqlStore,
-		log:      log.New("secrets.store"),
+func ProvideSecretsStore(db db.DB, bus bus.Bus) *SecretsStoreImpl {
+	store := &SecretsStoreImpl{
+		sb:               db,
+		log:              log.New("secrets.store"),
+		successCallbacks: new(sync.Map),
```

Comment:
I'm not super proud of this approach, but I tried some alternatives with no success. 
So, I'm completely open to any other suggestion.

I even tried to move the data keys cache management here, into the secrets store, but that would imply a much larger refactor and creates logical cycles (because it implies encryption/decryption vs database).